src/include/liburing.h

```diff
@@ -349,6 +361,9 @@ IOURINGINLINE void io_uring_cqe_seen(struct io_uring *ring,
 IOURINGINLINE void io_uring_sqe_set_data(struct io_uring_sqe *sqe, void *data)
 {
 	sqe->user_data = (unsigned long) data;
+#ifdef __SANITIZE_ADDRESS__
+	sqe->flags |= IOSQE_USER_DATA_IS_POINTER;
```

Comment:
Since this is strictly internal, it should not be exposed as if it's not. Particularly adding it to the official io_uring ABI header as well. I don't think we need to worry about the userdata pointer. If the application cares about that aspect, then it can just do its own sanitization on that. That's an app side thing imho, not something liburing should dip into. user_data should just remain an opaque type to both liburing and the kernel side.

Reply:
Ah, the reason I introduced this is because currently use_data is not being set to 0 in `io_uring_initialize_sqe`. Therefore, it is technically possible to request an sqe which still has a user_data set, which may at that point be invalid.

But I agree that given that this is app side solvable and can be removed.